### PR TITLE
Add api: get valid values for domain meta attributes 

### PR DESCRIPTION
**Pull request by OferLevi85**
To be used mainly by UI to display list of valid services for a user

### CODE REVIEW

File: core/zms/src/main/rdl/Domain.rdli [hunk 1]
```diff
@@ -282,3 +282,18 @@ resource DomainTemplate DELETE "/domain/{name}/template/{template}" {
         ResourceError TOO_MANY_REQUESTS;
     }
 }
+
+//List of Business Services
+type BusinessServiceList Struct {
```
Comment (havetisyan) [inline core/zms/src/main/rdl/Domain.rdli:287]: I don't see why we need to define a specific object type for Business Services. This is not an Athenz object - just list of strings so we should keep at as is - Array<String>, more details on the api call itself

File: core/zms/src/main/rdl/Domain.rdli [hunk 2]
```diff
@@ -282,3 +282,18 @@ resource DomainTemplate DELETE "/domain/{name}/template/{template}" {
...
+    Array<String> businessServices; //list of business services
+}
+
+//List all valid Business Services.
+resource BusinessServiceList GET "/bservices?user={user}" {
```
Comment (havetisyan) [inline core/zms/src/main/rdl/Domain.rdli:292]: We should also not have a specific endpoint for this - it assumes that somehow we're managing business services. We just need a more generic api if we want to implement such functionality. These are all domain meta attributes so it should be something like:

resource Array<String> GET "/domain/metastore?attribute=<attributeName>&user=<userName>

the server can verify that the attributeName is one of the supported meta store attributes (e.g. businessService, azureSubscription, awsAccount, etc. and call the appropriate megastore api.

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/metastore/DomainMetaStore.java [hunk 3]
```diff
@@ -45,6 +47,13 @@
      */
     void setBusinessServiceDomain(final String domainName, final String businessService);
 
+    /**
+     * Get a list of valid business services
+     * @param userName (optional) if not null, only get business services associated with the user
+     * @return Business Services List
+     */
+    List<String> getValidBusinessServices(final String userName);
```
Comment (havetisyan) [inline libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/metastore/DomainMetaStore.java:55]: we should also add similar api methods for the other attributes that the meta store supports:

getValidAWSAccounts()
getValidAzureSubscriptions(),e tc.

**Review by havetisyan — CHANGES_REQUESTED** [review]

File: core/zms/src/main/rdl/Domain.rdli (hunk 1 above)
Comment (OferLevi85) [reply core/zms/src/main/rdl/Domain.rdli]: I’m fixing it so the endpoint will be used to fetch valid values for attribute but I still won’t be able to return a list of strings as you showed in the example:

`resource Array<String> GET "/domain/metastore?attribute=<attributeName>&user=<userName>
`
RDL generator doesn’t accept it. The resource must be a declared type even if it’s just a String array (the name of the function generated in ZMSResources is determined by this type + action) So I’ve created the following type:
```
//List of valid domain meta attribute values
type DomainMetaStoreValidValuesList Struct {
    Array<String> validValues; //list of valid values for attribute
}
```

File: libs/java/server_common/src/test/java/com/yahoo/athenz/common/server/metastore/DomainMetaStoreTest.java [hunk 4]
```diff
@@ -41,6 +44,8 @@ public void testDomainMetaStoreDefaults() {
         assertTrue(metaStore.isValidBusinessService("athenz", "security"));
         assertTrue(metaStore.isValidBusinessService("athenz", null));
 
+        assertEquals(metaStore.getValidBusinessServices(null), new ArrayList<>());
```
Comment (havetisyan) [inline libs/java/server_common/src/test/java/com/yahoo/athenz/common/server/metastore/DomainMetaStoreTest.java:47]: we should add one of those for other attributes as well for code coverage.

**Review by havetisyan — CHANGES_REQUESTED** [review]

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java [hunk 5]
```diff
@@ -2375,6 +2375,52 @@ public void deleteDomainTemplate(ResourceContext ctx, String domainName, String
         dbService.executeDeleteDomainTemplate(ctx, domainName, templateName, auditRef, caller);
     }
 
+    @Override
+    public DomainMetaStoreValidValuesList getDomainMetaStoreValidValuesList(ResourceContext ctx, String attributeName, String userName) {
```
Comment (havetisyan) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java:2379]: Do we have test cases for this new method?

File: libs/java/server_common/src/test/java/com/yahoo/athenz/common/server/metastore/DomainMetaStoreTest.java (hunk 4 above)
Comment (OferLevi85) [reply libs/java/server_common/src/test/java/com/yahoo/athenz/common/server/metastore/DomainMetaStoreTest.java:47]: Added missing calls

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java (hunk 5 above)
Comment (OferLevi85) [reply servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java:2379]: Added missing tests

File: servers/zms/pom.xml [hunk 6]
```diff
@@ -39,7 +39,7 @@
   </dependencyManagement>
 
   <properties>
-    <code.coverage.min>0.96</code.coverage.min>
+    <code.coverage.min>0.964</code.coverage.min>
```
Comment (OferLevi85) [inline servers/zms/pom.xml:42]: Increased precision so we'll be alerted immediately on new functionality not covered with tests

**Review by havetisyan — APPROVED** [review]

**Pull request merged**